src/main/resources/static/js/custom/validator.js

```diff
@@ -0,0 +1,128 @@
+import { DynamicProperty } from "./dynamic-property.js";
+import { Constants } from "./global.js";
+
+export class Validator {
+
+    /** @type {Validator} */
+    static #instance;
+
+    /** @type {Validator} */
+    static get instance() {
+        return Validator.#instance ??= new Validator(Constants.HtmlElements.FORM_ELEMENT);
+    }
+
+    /** @type {HTMLFormElement} */
+    form;
+
+     /**
+      * The validation roster contains all the validations that need to be performed before the form is submitted.
+      * They are organised in a Map by HTMLElement (using the element's id as a key). 
+      * For each HTMLElement we maintain a list of the dynamic-properties that need to be validated.
+      * 
+      * There are two reasons we need to keep a roster:
+      * 1. So that we can trigger all validations without dispatching "change" events.
+      * 2. So that we can remove event handlers after an HTMLElement has been removed from the DOM.
+      * 
+      *  @type {Map<String, DynamicProperty[]>} 
+      */
+    validationRoster = new Map();
+
+    /**
+     * 
+     * @param {String} formElementId 
+     */
+    constructor(formElementId) {
+
+        if (Validator.#instance) {
+            throw new Error("You should not instantiate the Validator class yourself. It is mean to be used as singleton.")
+        }
+
+        this.form = document.getElementById(formElementId);
+        if (this.form === null) {
+            throw new Error(`Form element "${formElementId}" not found.`)
+        }
+
+        this.form.setAttribute("novalidate", "");
+        this.form.addEventListener("submit", Validator.onSubmit);
+    }
+
+    checkValidity() {
+        for(const elementId of this.validationRoster.keys()) {
+            this.checkValidityOfElement(document.getElementById(elementId))
+        }
+    }
+
+    checkValidityOfElement(htmlElement) {
+        if (htmlElement.readOnly || htmlElement.disabled) {
+            return;
```

Comment:
If an invalid value is loaded into a read only field (when we have notice loading later), it should still be validated.
Also anyone using a browser could still modify readOnly fields or some script could modify it (or modification via a bug).
Nothing urgent but we should keep this in mind.

Reply:
This is just a quick and dirty for us to avoid validating readonly fields in the metadata until the metadata is properly populated. That said, there is no point in doing live validation on a field that the user cannot change. If the data in the field are wrong then CVS will let the user know. Invalid data in a readonly field is an application bug; not a problem that the end user can fix.